hack/ci/setup-kind-cluster.sh

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Machine Controller Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source hack/lib.sh
+
+echodate "Setting up kind cluster..."
+
+if [ -z "${JOB_NAME:-}" ] || [ -z "${PROW_JOB_ID:-}" ]; then
+  echodate "This script should only be running in a CI environment."
+  exit 1
+fi
+
+export KIND_CLUSTER_NAME="${KIND_CLUSTER_NAME:-machine-controller}"
+
+start_docker_daemon_ci
+
+# Prevent mtu-related timeouts
+echodate "Setting iptables rule to clamp mss to path mtu"
+iptables -t mangle -A POSTROUTING -p tcp --tcp-flags SYN,RST SYN -j TCPMSS --clamp-mss-to-pmtu
```

Comment:
This is already part of the `start-docker.sh`: https://github.com/kubermatic/infra/blob/master/docker/build/start-docker.sh#L58-L59